Automattic/jetpack
# Autoloader: refactor functions.php

This PR is based on the `feature/update_autoloader` feature branch.

#### Changes proposed in this Pull Request:
* Refactor the functions in `functions.php` into the following classes:
  * `Autoloader_Handler` - used to find the latest autoloader.
  * `Classes_Handler` - used to set the class versions in the classmap
  * `Files_Handler` - used to set the file versions in the filemap.
  * `Plugins_Handler` - used to get information about the current plugin and the active plugins.

Note: In the next PR, a new class for determining which package version to use will be added. This class will be used by `Autoloader_Handler`, `Classes_Handler`, and `Files_Handler` to select which version of the autoloader, classes, and files to use.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* This changes the existing Autoloader package.

#### Testing instructions:
* Test the branch with plugins that use the autoloader and ensure that no errors are generated.

#### Proposed changelog entry for your changes:
* n/a
